src/nodes/cannonball.rs

```diff
@@ -0,0 +1,204 @@
+
+    const CANNONBALL_WIDTH: f32 = 32.;
+    pub const CANNONBALL_HEIGHT: f32 = 36.;
+    const CANNONBALL_ANIMATION_ROLLING: &'static str = "rolling";
+    const CANNONBALL_INITIAL_SPEED_X_REL: f32 = 600.;
+    const CANNONBALL_INITIAL_SPEED_Y: f32 = -200.;
+    //const CANNONBALL_MOUNT_X_REL: f32 = 20.;
+    //const CANNONBALL_MOUNT_Y: f32 = 40.;
+
+    const EXPLOSION_RADIUS: f32 = 4. * Self::CANNONBALL_WIDTH;
+
+    // Use Cannonball::spawn(), which handles the scene graph.
+    //
+    fn new(pos: Vec2, facing: bool, owner_id: u8) -> Self {
+        // This can be easily turned into a single sprite, rotated via DrawTextureParams.
+        //
+        let mut resources = storage::get_mut::<Resources>();
+        let cannonball_sprite = AnimatedSprite::new(
+            Self::CANNONBALL_WIDTH as u32,
+            Self::CANNONBALL_HEIGHT as u32,
+            &[Animation {
+                name: Self::CANNONBALL_ANIMATION_ROLLING.to_string(),
+                row: 0,
+                frames: 1,
+                fps: 1,
+            }],
+            true,
+        );
+
+        let speed = if facing {
+            vec2(
+                Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        } else {
+            vec2(
+                -Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        };
+
+        let mut body = PhysicsBody::new(
+            &mut resources.collision_world,
+            pos,
+            0.0,
+            vec2(Self::CANNONBALL_WIDTH, Self::CANNONBALL_HEIGHT),
+        );
+        body.speed = speed;
+
+        /*PhysicsBody {
+            pos,
+            facing,
+            angle: 0.0,
+            speed,
+            collider: None,
+            on_ground: false,
+            last_frame_on_ground: false,
+            have_gravity: true,
+            bouncyness: 1.0,
+        };*/
+
+        /*let cannonball_mount_pos = if facing {
+            vec2(Self::CANNONBALL_MOUNT_X_REL, Self::CANNONBALL_MOUNT_Y)
+        } else {
+            vec2(-Self::CANNONBALL_MOUNT_X_REL, Self::CANNONBALL_MOUNT_Y)
+        };*/
+
+        /*body.collider = Some(resources.collision_world.add_actor(
+            body.pos + cannonball_mount_pos,
+            CANNONBALL_WIDTH as i32,
+            CANNONBALL_HEIGHT as i32,
+        ));*/
+
+        Self {
+            cannonball_sprite,
+            body,
+            lived: 0.0,
+            countdown: Self::CANNONBALL_COUNTDOWN_DURATION,
+            owner_id,
+            owner_safe_countdown: Self::CANNONBALL_OWNER_SAFE_TIME,
+            //erupting: false,
+            //erupting_enable_on_y: None,
+        }
+    }
+
+    pub fn spawn(pos: Vec2, facing: bool, owner_id: u8) {
+        let cannonball = Cannonball::new(pos, facing, owner_id);
+        scene::add_node(cannonball);
+    }
+}
+
+/*impl EruptedItem for Cannonball {
```

Comment:
Same situation here

If the code is temporary turned off for the sake of PR - it may be worth doing an additional draft PR with said code, but keeping the PR itself clean.

In some cases it really is easier to just comment out something, but, at least, leave a comment on why the code is commented (and I feel that right now this is not the case, and commented code just do not belong to the feature implemented by this PR)